src/Microsoft.SqlTools.ServiceLayer/QueryExecution/QueryExecutionService.cs

```diff
@@ -539,6 +511,50 @@ private static void ExecuteAndCompleteQuery(QueryExecuteParams executeParams, Re
             }
         }
 
+        // Internal for testing purposes
+        internal string GetSqlText(ExecuteRequestParamsBase request)
```

Comment:
After looking at this some more, I think the changes required are going to be too massive for this code review (large changes to connection service). Since I want to get moving on the Edit Data stuff, I'm going to decline to implement these changes for now and stick with the slightly less than ideal implementation that's currently in review. I have most of it written up (see https://github.com/Microsoft/sqltoolsservice/tree/fix/queryConnectionIsolation), but I want to make sure that I get the unit tests right -- which will likely take a fair amount of time.